omniduct/filesystems/s3.py

```diff
@@ -0,0 +1,154 @@
+import os
+import re
+
+import pandas as pd
+
+from .base import FileSystemClient
+from ..utils.debug import logger
+
+
+class S3Client(FileSystemClient):
+
+    PROTOCOLS = ['s3']
+    DEFAULT_PORT = 80
+
+    def _init(self, bucket=None, aws_profile=None):
+        assert bucket is not None, 'S3 Bucket must be specified using the `bucket` kwarg.'
+        self.bucket = bucket
+        self.aws_profile = aws_profile
+
+        self._setup_session()
+
+    def _setup_session(self):
+        import boto3
+        import botocore
+        s = boto3.Session(profile_name=self.aws_profile)
+        self._session = s
+        self._client = s.client('s3')
+        self._resource = s.resource('s3')
+        self.host = 'autoscaling.{}.amazonaws.com'.format(self._client.meta.region_name)
+
+    def _connect(self):
+        # Refresh access token
+        from opinel.utils.credentials import read_creds, init_sts_session, read_creds_from_aws_credentials_file
+        credentials = read_creds(self.aws_profile)
```

Comment:
if the main point is that read_creds changes the state perhaps don't assign this to a variable for clarity

Reply:
Good point. Saved output to `self._credentials` for debugging, but commented on its changing of "state" (it updates the access token in your ~/.aws/credentials file if you are accessing AWS services using such configuration).